src/server/api/redirect.ts

```diff
@@ -171,18 +180,52 @@ export default async function redirect(req: Express.Request, res: Express.Respon
     if (gaTrackingId) gaLogging(req, res, shortUrl, longUrl)
 
     // Redirect immediately if a crawler is visiting the site
-    if (isCrawler(req.headers["user-agent"] || '')) {
+    if (isCrawler(req.headers['user-agent'] || '')) {
       res.status(302).redirect(longUrl)
       return
-    } else {
-      // Serve transition page
-      // TODO: Get/set a browser cookie so that this isn't served too frequently
+    }
+
+    // Reassure typescript that the session object exists.
+    if (!req.session) {
```

Comment:
we should simply use `req.session!.property`, instead of modifying our program's behavior and writing code for the compiler.